internal/jobcontainers/logon.go

```diff
@@ -0,0 +1,64 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+// Takes in a DOMAIN\Username or just Username and will return a token
+// for the account if successful.
+func processToken(user string) (windows.Token, error) {
+	var (
+		domain   string
+		userName string
+		token    windows.Token
+	)
+
+	split := strings.Split(user, "\\")
+	if len(split) == 2 {
+		domain = split[0]
+		userName = split[1]
+	} else if len(split) == 1 {
+		userName = split[0]
+	}
```

Comment:
Should we error out if we get something that is not len or 1 or 2 or empty?

Reply:
I suppose. I dont believe cexecsvc handles that case either and it will just error out when trying to grab a token but I can add if you'd prefer

Reply:
I think it would be more complete to add it here

Reply:
Sure